code/modules/projectiles/ammo_datums.dm

```diff
@@ -332,7 +332,11 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	sundering = 3
 
 /datum/ammo/bullet/revolver/on_hit_mob(mob/M,obj/projectile/P)
-	staggerstun(M, P, stagger = 1, slowdown = 0.5, knockback = 1, shake = 0.5)
+	var/obj/item/weapon/gun/shot_from = P.shot_from
+	var/stagger_knock = 0 // snub the snubnose stagger
+	if(shot_from.attachments?.Find(ATTACHMENT_BARREL_MOD))
+		stagger_knock = 1
+	staggerstun(M, P, stagger = stagger_knock, slowdown = 0.5, knockback = stagger_knock)
```

Comment:
actually looking at this, this nerfs all revolvers when used normally since they don't have snub nose versions, think to do that without touching them would need to a standard revolver specific bullet.

Reply:
No problem, I made it check to make sure the weapon can take that attachment also or it defaults to having it.